scylla/src/transport/cluster.rs

```diff
@@ -66,6 +68,65 @@ struct ClusterWorker {
     used_keyspace: Option<VerifiedKeyspaceName>,
 }
 
+struct Keepaliver {
+    cluster_data: Arc<ArcSwap<ClusterData>>,
+    keepalive_interval: Duration,
+}
+
+impl Keepaliver {
+    pub fn get_data(&self) -> Arc<ClusterData> {
+        self.cluster_data.load_full()
+    }
+
+    async fn work(self) {
+        let mut interval = tokio::time::interval(self.keepalive_interval);
+        interval.tick().await; // Use up the first tick.
+
+        // Default behaviour (Burst) is not suitable for sending keepalives.
+        use tokio::time::MissedTickBehavior::Delay;
+        interval.set_missed_tick_behavior(Delay);
+
+        loop {
+            interval.tick().await;
+
+            let data = self.get_data();
+            let nodes = data.get_nodes_info();
+            for node in nodes {
+                Self::keepalive_node_connections(node).await;
+            }
+        }
+    }
+
+    async fn send_keepalive_query(connection: &Connection) {
+        // FIXME timeout?
```

Comment:
not explicitly - on each tick, let's kill any hanging requests that didn't get a response in time and send new ones